### ISSUE TITLE
Device identifier: Not available during ISingleAccountPublicClientApplication signIn

### ISSUE THREAD

**Issue by rajohns08** [AzureAD/microsoft-authentication-library-for-android#2136]
**Describe the bug**
My app is live and in production. I see the below screenshot after signing in using the `ISingleAccountPublicClientApplication` `signIn` method. My device is already registered in Intune despite the message saying otherwise. The Azure backend dev on my team confirmed our conditional access policy is failing because they don't receive a device ID.

**Smartphone (please complete the following information):**
 - Device: This happens on every device I've tried but the most recent device is a Samsung Galaxy A20
 - Android Version: 11, 12, 14
 - Browser: This is an in-app webview per my msal configuration
 - MSAL Version: 4.0.5

**To Reproduce**
Simply call the `ISingleAccountPublicClientApplication` `signIn` method. After entering password and mfa code in the WebView, the screenshot below appears.

If related to development, please provide relevant configuration details necessary to understand your problem including any relevant traces, logs, or otherwise.

Config file (I have also tested authorization_user_agent set to BROWSER as well, with default set to Chrome and Edge):

```
{
  "client_id" : "<redacted>",
  "authorization_user_agent" : "WEBVIEW",
  "redirect_uri" : "<redacted>",
  "broker_redirect_uri_registered" : true,
  "authorities" : [
    {
      "authority_url": "<redacted>",
      "type": "B2C",
      "audience": {
        "tenant_id": "myidaasuat.onmicrosoft.com",
        "type": "AzureADandPersonalMicrosoftAccount"
      }
    }
  ],
  "account_mode": "SINGLE"
}
```

**Expected behavior**
Device ID is passed to backend, and I don't see this message telling me my device isn't registered. Signin works.

**Actual Behavior**
Device ID isn't passed to backend, and I see this message saying my device isn't registered. 

**Screenshots**
![msal2](https://github.com/AzureAD/microsoft-authentication-library-for-android/assets/2711642/5a5cf4ed-f215-49e1-907e-ad8ee69ccd2d)

**Additional context**
I enabled MSAL logging and got the below log file:
[logs.txt](https://github.com/user-attachments/files/16087566/logs.txt)

**Please note: Do not include sensitive information like PII, OII, credentials, secrets, and tokens.**

For **privacy/security** issues please see instructions [here](https://github.com/AzureAD/microsoft-authentication-library-for-android#security-reporting)

**Comment by SuperCaliMan** [comment]
Hi, same problem, have you found any solutions?

**Comment by negoe** [comment]
hi @rajohns08, This is an expected behavior. The Conditional Access device compliance feature isn't available in Azure AD B2C tenants. Please read these documentation on[ Azure AD B2C service limitation](https://learn.microsoft.com/en-us/azure/active-directory-b2c/conditional-access-user-flow?source=recommendations&pivots=b2c-user-flow#service-limitations-and-considerations) and [CA Policies for Azure AD B2C](https://learn.microsoft.com/en-us/azure/active-directory-b2c/conditional-access-identity-protection-overview).

**Comment by rajohns08** [comment]
@negoe I'm not sure what you mean exactly, but I got this working? I needed to remove the `"authorization_user_agent" : "WEBVIEW"` line from my MSAL config, so SSO would go through brokered authentication: https://learn.microsoft.com/en-us/entra/msal/android/single-sign-on#sso-through-brokered-authentication. Also, ironically, it only works when Chrome is set as the default browser, not Edge.

cc: @SuperCaliMan
